pkgs/development/python-modules/linien-common/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, setuptools
+, importlib-metadata
+, numpy
+, rpyc
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "linien-common";
+  version = "0.8.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "linien-org";
+    repo = "linien";
+    rev = "v${version}";
+    hash = "sha256-JYXJEa3HerBTYzdr6AK6tWVa5auVy4REsTpO9j1d/HE=";
+  };
+
+  sourceRoot = "source/linien-common";
+
+  nativeBuildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [
+    importlib-metadata
+    numpy
+    rpyc
+    scipy
+  ];
+
```

Comment:
Whats with tests and the pythonImportsCheck?

Reply:
`pythonImportsCheck` added

Reply:
I think we can add [these tests](https://github.com/linien-org/linien/tree/master/tests) like ipykernel. 
https://github.com/NixOS/nixpkgs/tree/nixos-unstable/pkgs/development/python-modules/ipykernel

Reply:
Hmm can you help provide more details how to create the `tests.nix` file here? Like what dependencies I should use.

Reply:
@natsukium

Reply:
Sorry for the late reply.

We need to add pytestCheckHook, linien-common and linien-client to nativeCheckInputs.
Also, looking at some test files, it looks like numpy and scipy are needed as well.

Regarding the test packages, ipykernel's tests.nix would be helpful, and you can use buildPythonPackages with `pyproject = false`.
The src could be shared with linien-common.

Reply:
@natsukium updated

Reply:
> This is because this repository has a monorepo like structure. linien-common, linien-client, ... packages are in the subdirectory, but tests is defined at the top level.

Thanks for the explanation.

Marked as resolved.